src/counter.cc

```diff
@@ -17,20 +17,27 @@
 namespace benchmark {
 namespace internal {
 
-double Finish(Counter const& c, double cpu_time, double num_threads) {
+double Finish(Counter const& c, int64_t iterations, double cpu_time,
+              double num_threads) {
   double v = c.value;
   if (c.flags & Counter::kIsRate) {
     v /= cpu_time;
   }
   if (c.flags & Counter::kAvgThreads) {
```

Comment:
Should i follow up with `kIsThreadInvariant`, for symmetry?

Reply:
After thinking about it a bit more, it would make no sense, since that is already what happens.
https://github.com/google/benchmark/blob/master/README.md#user-defined-counters
```
In multithreaded benchmarks, each counter is set on the calling thread only. When the benchmark finishes, the counters from each thread will be summed; the resulting sum is the value which will be shown for the benchmark.
```

BUT i will have to follow-up with `*= cpu_time;`
Now, if the benchmark processes `N` Things (not `items`, more fine-grained),
`kIsRate` would tell you how many Things it processes per second.
But what about how many seconds per `Thing`?

In the same thought, i wonder how `Set{Bytes,Items}Processed()` are useful
given the existance of custom counters. I wonder whether they should be
deprecated v2. One very obvious downside is that they have very generic names,
and it is impossible, unlike custom counters, to customize them.